gpflow/experimental/check_shapes/parser.py

```diff
@@ -204,13 +239,44 @@ def _argument_spec_to_sphinx(self, argument_spec: ParsedArgumentSpec) -> str:
         out.append(f"* **{repr(argument_spec.argument_ref)}**")
         out.append(" has shape [")
         out.append(self._shape_spec_to_sphinx(shape_spec))
-        out.append("].")
+        out.append("]")
+
+        if argument_spec.condition is not None:
+            out.append(" if ")
+            out.append(self._bool_spec_to_sphinx(argument_spec.condition, False))
+
+        out.append(".")
+
         if tensor_spec.note is not None:
             note_spec = tensor_spec.note
             out.append(" ")
             out.append(note_spec.note)
         return "".join(out)
 
+    def _bool_spec_to_sphinx(self, bool_spec: ParsedBoolSpec, paren_wrap: bool) -> str:
```

Comment:
(you could do the same trick here if you'd like to avoid unnecessary brackets in "a or b or c" or "a and b or c" etc)

Reply:
Again, I think I like the extra brackets. I think they add readability.